lib/workers/mixins/web_server_worker_mixin.rb

```diff
@@ -14,4 +14,32 @@ def do_before_work_loop
     # Since puma/thin traps interrupts, log that we're going away and update our worker row
     at_exit { do_exit("Exit request received.") }
   end
+
+  module ClassMethods
+    def start_worker(*args)
+      worker = self.new(*args)
+      _log.info("URI: #{worker.worker.uri}")
+
+      # Do all the SQL worker preparation in the main thread
+      worker.prepare
+
+      # TODO: Need to rename build_command_line to options_hash or something
+      options = corresponding_model.build_command_line(:Port => args.first[:Port])
+
+      # The heartbeating will be done in a separate thread
+      Thread.new { worker.run }
+
+      start_rails_server(options)
+    end
+
+    def start_rails_server(options)
+      require "rails/commands/server"
+
+      _log.info("With options: #{options.inspect}")
+      Rails::Server.new(options).tap do |server|
+        Dir.chdir(Vmdb::Application.root)
+        server.start
```

Comment:
Do what `rails server` does at the command line minus all of the ARGV translations.